orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/metrics/AtlasQueueMonitor.kt

```diff
@@ -71,6 +88,27 @@ class AtlasQueueMonitor
     _lastState.set(queue.readState())
   }
 
+  @Scheduled(fixedDelayString = "\${queue.zombie.check.frequency:3600000}")
+  fun checkForZombies() {
+    val startedAt = clock.instant()
+    repository.retrieve(PIPELINE)
+      .mergeWith(repository.retrieve(ORCHESTRATION))
+      .subscribeOn(zombieCheckScheduler.orElseGet(Schedulers::io))
+      .filter { it.status == RUNNING }
+      .filter(this::hasBeenAroundAWhile)
+      .filter(this::queueHasNoMessages)
+      .doOnCompleted {
+        log.info("Completed zombie check in ${Duration.between(startedAt, clock.instant())}")
+      }
+      .subscribe {
+        val tags = mutableListOf<Tag>(BasicTag("application", it.application))
+        if (it.type == PIPELINE && it.pipelineConfigId != null) {
+          tags.add(BasicTag("name", it.name))
+        }
+        registry.counter("zombie.execution", tags).increment()
```

Comment:
Tag with application and pipeline name if it's a pipeline that exists in fiddy

Reply:
I'd suggest not including the pipeline name ... that's potentially going to blow up big